src/mscorlib/shared/System/IO/Path.Windows.cs

```diff
@@ -149,6 +159,20 @@ public static string GetPathRoot(string path)
             return pathRoot <= 0 ? string.Empty : path.Substring(0, pathRoot);
         }
 
+        public static ReadOnlySpan<char> GetPathRoot(ReadOnlySpan<char> path)
+        {
+            ReadOnlySpan<char> result = new ReadOnlySpan<char>();
```

Comment:
inline this, it's only used in one place, then you won't new it up except in the case you need it.